src/e2e-test/java/io/cdap/plugin/utils/PubSubClient.java

```diff
@@ -56,4 +101,92 @@ public static String getTopicCmekKey(String topicId) throws IOException {
     return getTopic(topicId).getKmsKeyName();
   }
 
+
+  public static void publishWithErrorHandlerExample(String projectId, String topicId, List<String> dataMessages)
+    throws IOException, InterruptedException {
+    TopicName topicName = TopicName.of(projectId, topicId);
+    Publisher publisher = null;
```

Comment:
Instead of creating a new Publisher instance for each message, can you reuse the same Publisher instance for all messages. Creating a Publisher is an expensive operation, and reusing it can significantly improve performance.

Reply:
As Discussed we already have the after hooks for deletion of topics and subscription